parser.c

```diff
@@ -704,16 +704,6 @@ void process_pihole_log(int file)
 			validate_access_oTfd(timeidx, forwardID, __LINE__, __FUNCTION__, __FILE__);
 			overTime[timeidx].forwarddata[forwardID]++;
 		}
-		else if((strstr(readbuffer,"IPv6") != NULL) &&
```

Comment:
I remove this code as it was always a dirty hack to detect whether `dnsmasq` was restarted. We cannot use it to detect a `reload` condition. As gravity will now always send a `SIGHUP`, we can safely remove this code here.

Reply:
How about the case where the user manually restarts dnsmasq such as through `sudo service dnsmasq restart`?

Reply:
`FTL` doesn't need to know if/when `dnsmasq` is restarted, strictly speaking. We used the fact that `dnsmasq` is restarted in the past to decide that the user may have changed something (updated gravity, added/removed blacklist entries, etc.) that needs our attention. This PR, however, implements a direct communication channel between our core code and `FTL` through which is will be "informed" when it should re-parse the gravity files. If the user just does a manual `restart`, we don't actually have to re-parse the lists as it is highly unlikely that the analyzed lists have changed. Hence, it is okay if we miss these restarts as they contain no information that would be of value to `FTL`.